clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -461,7 +499,64 @@ public Path getWorkingDirectory() {
 
     @Override
     public boolean mkdirs(Path path, FsPermission fsPermission) throws IOException {
-        return true;
+        try {
+            // check path is not a directory already
+            FileStatus fileStatus = getFileStatus(path);
+            if (fileStatus.isDirectory()) {
+                return true;
+            }
+            throw new FileAlreadyExistsException("Path is a file: " + path);
+        } catch (FileNotFoundException e) {
+            // check if part of path is a file already
+            ObjectLocation objectLocation = pathToObjectLocation(path);
+            Path branchRoot = new Path(String.format("%s://%s/%s", Constants.URI_SCHEME, objectLocation.getRepository(), objectLocation.getRef()));
```

Comment:
Why do we glue a Path like this rather than use e.g. `ObjectLocation.formatPath`?